olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/StreamLayerClient.h

```diff
@@ -72,7 +72,11 @@ class StreamLayerClientImpl;
  * auto request =
  * olp::dataservice::read::SubscribeRequest().WithSubscriptionMode(
  *             olp::dataservice::read::SubscribeRequest::SubscriptionMode::kSerial));
- * auto cancellable_future = stream_client.Subscribe(request);
+ *
+ * auto subscribe_response = stream_client.Subscribe(request).GetFuture().get();
+ * if (response.IsSuccessful()) {
+ *   auto unsubscribe_response = stream_client.Unsubscribe().GetFuture().get();
+ * }
```

Comment:
This is not really a common use case and you confuse the user with this.
What you are telling me is this: I should subscribe to the stream layer and if successful then unsubscribe directly.
Instead what you want to do it this:

```cpp
auto request = olp::dataservice::read::SubscribeRequest().WithSubscriptionMode(
    olp::dataservice::read::SubscribeRequest::SubscriptionMode::kSerial));
auto subscribe_response = stream_client.Subscribe(request).GetFuture().get();
if (response.IsSuccessful()) {
    // Successfully subscribed, now you can consume data from the stream layer 
}

// Consume data from the stream layer, seek offsets, etc.

// After you are finished and you want to stop consumption unsubscribe
auto unsubscribe_response = stream_client.Unsubscribe().GetFuture().get();
if (unsubscribe_response.IsSuccessful()) {
    // Successfully unsubscribed.
}
```